homeassistant/components/network/websocket.py

```diff
@@ -62,3 +69,28 @@ async def websocket_network_adapters_configure(
         msg["id"],
         {ATTR_CONFIGURED_ADAPTERS: network.configured_adapters},
     )
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "network/url",
+        vol.Required("url_type"): vol.In(URL_TYPES),
+    }
+)
+@websocket_api.async_response
+async def websocket_network_url(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Get the internal URL."""
+    connection.send_result(
+        msg["id"],
+        get_url(
```

Comment:
Should we make this websocket command a bit more generally useful ? Maybe we can just return all URLs for all situations as a dict?

```
{ "internal": "…", "external": "…", "cloud": "…" }
```